### DISCUSSION TITLE
Module dependency analysis for development environments

### DISCUSSION THREAD

**Discussion by hyzx86** [OrchardCMS/OrchardCore#13612] [General]
When developing OrchardCore application,
I often encountered the problem that modules are in dependent order. I wonder if anyone has made a similar tool before.

Is there a better way to analyze the order of dependencies between modules?

Like logging when a module is loaded?

**Comment by hishamco** [comment]
When you enable any feature OC already enable the module dependency, could you please elaborate where's the problem exactly?

**Comment by hyzx86** [comment]
Mainly used for checking module loading order, in regular development. VisualStudio will automatically check for dependency issues.
But there's no way to analyze dependencies more intuitively in OC

**Comment by hishamco** [comment]
> But there's no way to analyze dependencies more intuitively in OC

If there's an issue we can think about a solution

**Comment by ns8482e** [comment]
Are you referring about nuget packages and references at build time or module's dependency defined in manifest for runtime?

In both cases I didn't find any issue in dev

**Comment by hyzx86** [comment]
> Are you referring about nuget packages and references at build time or module's dependency defined in manifest for runtime?
> 
> In both cases I didn't find any issue in dev

 Just like this:
Manifest.cs
![image](https://user-images.githubusercontent.com/15613121/229802624-2ea928b2-8c70-47d0-9751-897a532fdefb.png)
Startup.cs
![image](https://user-images.githubusercontent.com/15613121/229802687-b5ff9128-47cd-4add-8a95-a31a860f3204.png)


In many cases, it is necessary to observe the dependencies manually, and if the module is loaded after the dependent module, it will cause the interface to not be overridden or  dependency injection doesn't work

**Comment by ns8482e** [comment]
If you have dependency defined, dependency loaded before the dependent module
